test/test_bot.py

```diff
@@ -308,51 +309,52 @@ def url_handler(*args, **kwargs):
     assert not list(results), 'URL must not match any pattern'
 
 
-def test_register_url_callback_twice(tmpconfig):
-    """Test register_url_callback replace URL callbacks for a pattern."""
+def test_register_url_callback_multiple(tmpconfig):
+    """Test register_url_callback adding multiple callbacks for a pattern."""
```

Comment:
This patch doesn't enable multiple callbacks for a pattern yet.

Reply:
It doesn't, but the test still tests it. This test's behavior was reverted (in this branch) to the old test's behavior of ensuring that the second-added clobbered the first-added.

Reply:
Fair enough on future-proofing the test name, I suppose. But there are a lot of changes below given that no behavior has changed. I don't want to waste time now reviewing stuff that doesn't apply to what we're shipping in 7.0.